src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -480,12 +505,60 @@ private static bool TryParseGuid(ReadOnlySpan<char> guidString, ref GuidResult r
                         TryParseExactN(guidString, ref result),
             };
         }
+        private static bool TryParseGuid(ReadOnlySpan<byte> guidString, ref GuidResult result)
+        {
+            // Remove whitespace from beginning and end
+            if (guidString.Length != 0 && (IsWhite(guidString[0]) || IsWhite(guidString[^1])))
+            {
+                guidString = Trim(guidString);
+            }
+
+            if (guidString.Length < 32) // Minimal length we can parse ('N' format)
+            {
+                result.SetFailure(ParseFailure.Format_GuidUnrecognized);
+                return false;
+            }
 
-        private static bool TryParseExactB(ReadOnlySpan<char> guidString, ref GuidResult result)
+            return (guidString[0]) switch
+            {
+                (byte)'(' => TryParseExactP(guidString, ref result),
+                (byte)'{' => guidString[9] == (byte)'-' ?
+                        TryParseExactB(guidString, ref result) :
+                        TryParseExactX(guidString, ref result),
+                _ => guidString[8] == (byte)'-' ?
+                        TryParseExactD(guidString, ref result) :
+                        TryParseExactN(guidString, ref result),
+            };
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            static ReadOnlySpan<byte> Trim(ReadOnlySpan<byte> span)
+            {
+                int start = 0;
+                for (; start < span.Length; start++)
+                {
+                    if (!IsWhite(span[start]))
```

Comment:
This is questionable. UTF8 text should not be manipulated in byte like this. It's not ASCII.
If we do intend to trim ASCII whitespaces only, use `Ascii.Trim`.